andela/Ah-backend-guardians
# #162685678 Users should be able to report an article

This PR enables authorised users to report an article

#### Description of Task to be completed?
As an authenticated user, I should be able to report articles that violate terms of the agreement or are plagiarised to the authors haven admin for removal and or any other actions as per the terms of engagement

#### What are the relevant Pivotal Tracker stories?
#162685678[https://www.pivotaltracker.com/n/projects/2232496]

### How should this be manually tested?
1. Run each of the following commands in your terminal in the respective order.

* git clone https://github.com/andela/Ah-backend-guardians.git
* view the cloned repo in VS code
* In the terminal of VSCode, git checkout ft-report-an-article-162685678
* Activate the virtual environment. (.venv/bin/activate)
* Run the app python manage.py runserver
* Lauch postman
* Hit end point http://127.0.0.1:8000/api/users/ to register a user
  use parameters
  {
  "username":"pault",
  "email": "[paulviv1@gmail.com](mailto:paulviv1@gmail.com)",
  "password":"nabuloff123W"
  }
* Hit end point http://127.0.0.1:8000/api/users/login to login a user
  {
  "email": "[paulviv1@gmail.com](mailto:paulviv1@gmail.com)",
  "password":"nabuloff123W"
  }
* Authenticate the user by passing the generated token in the header or place the token in the environment variables
  Hit this URL to access the create report endpoint. post http://127.0.0.1:8000/api/articles/slug/report/
  Use these parameters
  {
  "reason": "Tribalistic"
  }
  You should get this result .  

![image](https://user-images.githubusercontent.com/41106626/53652290-95883b80-3c59-11e9-892b-44029b26bc00.png)

* Hit this URL to access the a list of reported articles by the current authorised user.
  get http://127.0.0.1:8000/api/articles/slug/report/
* Hit this URL to access the getReportedArticles by a non-staff member endpoint.
  get http://127.0.0.1:8000/api/articles/report/<username>/

You should get this result

![image](https://user-images.githubusercontent.com/41106626/53652166-4215ed80-3c59-11e9-8304-12d5eb85a721.png)
